### DISCUSSION TITLE
Where i should add these macros?

### DISCUSSION THREAD

**Discussion by intoinside** [c128lib/.github#6] [Team Posts]
Hi developers!
I've added some issue on common (https://github.com/c128lib/common/issues/13 https://github.com/c128lib/common/issues/11 for embedding sort algorithm macros and https://github.com/c128lib/common/issues/12 for a random number generator)

It doesn't sounds good to common project, maybe these are better to framework?

I would add a specific module for sorting algorithms (sort.asm and sort-global.asm) and one for random numbers (random.asm and random-global.asm). They're not chipset items, as they don't use a particular chip or expose a feature, and common also doesn't seem right to me because these are pretty elaborate features and don't define any new label.

What do you think? Should they be moved to framework?

**Comment by johnpalermo** [comment]
I was thinking the same, to add to framework.

**Comment by intoinside** [comment]
ok... issue moved, i started working on bubble sort on separate branch.
I've also added CI integration so every push to master will trigger a build on circle ci. You can see it by that green sign

![image](https://user-images.githubusercontent.com/8144124/215892960-1de022d5-9f55-4189-886a-bf0fe4405f12.png)

**Comment by johnpalermo** [comment]
I started working on the string library on a separate branch. I’m not really too familiar with gradle or circleci. On my local repository, KickAssembler cannot resolve the paths to included libs. I will attach screenshots later. I’m on my mobile now. I also see some assert statements in the code. Do you have a specific workflow for building and testing? Grazie.

**Comment by intoinside** [comment]
To use Gradle, you just need to ensure that you have JDK 11 or better. You can download it here https://jdk.java.net/
Gradle is already included in repository code so you must open a terminal window on your local repository and then type "gradlew build". This command launch a build over you project, it uses your build.gradle configuration file to check directories and dependencies. This should solve your problems about path resolving.
At the end, it tells you if there are compilation error on your code.
Usually i create assert for macros to ensure that code generated by macro is correct, but i'll always check if with a test assembly code. Assert and AssertError are executed every "gradlew build" command and only test statically your code so it's good but it's not the best.
CircleCi is automatically called for every push done on GitHub on master branch. I'm using a free plan so i prefer to limit only official branch.

**Reply by johnpalermo** [reply]
Ok.  I see that Maciej Małecki created his own gradle plugin to auto build with KickAssember.  Very nice.

So now after running gradlew my paths are resolved.  Thanks.  I also see the assert output after the build.
